src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.tsx

```diff
@@ -279,15 +279,31 @@ function PopoverReportActionContextMenu(_props: unknown, ref: ForwardedRef<Repor
         setIsPopoverVisible(false);
     };
 
+    const [transactionIDs, setTransactionIDs] = useState<string[]>([]);
+
+    useEffect(() => {
+        const reportAction = reportActionRef.current;
+        if (isMoneyRequestAction(reportAction)) {
+            const originalMessage = getOriginalMessage(reportAction);
+            if (originalMessage && 'IOUTransactionID' in originalMessage && !!originalMessage.IOUTransactionID) {
+                setTransactionIDs([originalMessage.IOUTransactionID]);
+                return;
+            }
+        }
+        setTransactionIDs([]);
+    }, [reportActionRef]);
```

Comment:
Adding ref here has no effect because it won't trigger re-render.

Checking the commit history, I saw you first used:
```javascript

const IOUTransactionID = originalMessage && 'IOUTransactionID' in originalMessage ? originalMessage.IOUTransactionID : undefined;
```

What's the reason for this useEffect?